benchmark/google/view_transpose_impl.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright (C) 2020 Samuel Debionne, ESRF.
+
+// Use, modification and distribution is subject to the Boost Software
+// License, Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#include <benchmark/benchmark.h>
+
+#include <boost/gil.hpp>
+
+#define GIL_ENABLE_UNROLLED 1
+
+namespace boost {
+    namespace gil {
+
+        template <typename SrcView, typename DstView>
+        static void transpose(const SrcView& src, const DstView& dst)
+        {
+            int i, j;
+            auto m = src.width();
+            auto n = src.height();
+
+            static const int block_size = 4;
+
+            std::array<gray8c_view_t::xy_locator::cached_location_t, block_size* block_size> l;
+            std::generate(l.begin(), l.end(), [loc = src.xy_at(0, 0), i = 0]() mutable {
+                auto res = loc.cache_location(i % block_size, i / block_size);
+                i++;
+                return res;
+            });
+
+#if GIL_ENABLE_UNROLLED
```

Comment:
We should prefix all our macros with `BOOST_GIL_*`, see https://github.com/boostorg/gil/issues/410

I'd suggest to rename this to `BOOST_GIL_BENCHMARK_ENABLE_UNROLLED` or similar.